### ISSUE TITLE
`aws_codebuild_report_group` needs time to delete previously generated reports, fails `terraform apply`

### ISSUE THREAD

**Issue by jessefarinacci** [hashicorp/terraform-provider-aws#17290]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_codebuild_report_group

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "aws_kms_alias" "s3" {
  name = "alias/aws/s3"
}

resource "aws_codebuild_report_group" "tests" {
  name  = "myreport-tests"
  type  = "TEST"

  export_config {
    type = "S3"

    s3_destination {
      bucket              = "my-bucket-name"
      encryption_disabled = false
      encryption_key      = data.aws_kms_alias.s3.arn
      packaging           = "NONE"
      path                = "/reports"
    }
  }
}
```

### Debug Output

```
Error: error deleting CodeBuild Report Groups(arn:aws:codebuild:xxx:yyy:report-group/zzz):
InvalidInputException: Report group could only be deleted after its reports are deleted
```

### Panic Output

N/A

### Expected Behavior

Terraform should poll/wait for reports to be deleted and the report group to be deleted before moving forward in the lifecycle.

### Actual Behavior

Terraform fails to delete the report group immediately because existing reports are still being deleted, and then the `terraform apply` fails.. if I wait a minute for AWS to clean up the reports/report-group then re-run the apply, it succeeds.

### Steps to Reproduce

1. `terraform apply`
2. create some reports via codebuild
3. make a change to the report-group.tf, e.g. name change `tests` -> `tests2`
4. `terraform apply`
5. tf will see recreate required, but fail

### Important Factoids

N/A

### References

N/A

**Comment by bflad** [comment]
@DrFaust92 heads up that I'm relabeling this as an enhancement to support for the [`DeleteReports` parameter for the `DeleteReportGroup` API operation](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_DeleteReportGroup.html) -- the Terraform resource is currently doing the correct (safer) operation of not deleting practitioner data without consent.

**Comment by bflad** [comment]
Support for a `delete_reports` argument in the `aws_codebuild_report_group` resource has been merged and will release with version 3.28.0 of the Terraform AWS Provider, later today. Thank you to @DrFaust92 for the implementation. 👍
